TU20Bot/Models/EventModel.cs

```diff
@@ -57,5 +62,7 @@ public class EventModel {
 
         // Used by MongoDB to sort.
         public double? textScore;
+
+        public EventState state;
```

Comment:
I think this would be more nicely represented as `bool isDraft;`

Reply:
Is there a case where a draft message does not have a prompt? I would assume I thought this through, but it's possible I didn't.

If there is not a case where a draft message does not have a prompt, you should use the nullable prompt variable instead. Calculations should already be handled by that :) If not though, go crazy I guess.

Reply:
I will be leaving the use of the enum in its current form as this would be a syntactical change to achieve a similar result.

Yes you did think it through! For both drafts and newly confirmed messages there will always be a non-null promptId value in the database, meaning that without additional data or adding differentiating logic there will not be a way to differentiate them.
You built an elegant solution that works well for handling this. In three cases (confirmation, cancellation, deletion of prompt), the relevant promptId will always be set in the database until its removal is handled by `messageDeleted` as the method responsible for all the case cleanups.

Reply:
> I will be leaving the use of the enum in its current form as this would be a syntactical change to achieve a similar result.

I think code quality and readability is impacted. Enum implies that there are many states to pick from. The ideal type for this variable would be a boolean. It's also more nicely represented in bson and by-hand database queries.

> For both drafts and newly confirmed messages...

I'm getting the impression `isDraft` is a temporary value? A confirmed message should have a null `promptId`. Please provide a good reason for the additional complexity.

Reply:
Any impacts to code quality and readability are subjective as both are clear to interpret; this is also true in the current implementation as it is immediately clear to understand what the code is doing and to write additional comprehensive logic. Good news, I have updated the implementation to use a boolean value!

Both drafts and newly confirmed messages have non-null promptId values. Previously confirmed messages have a null promptId value. This behaviour is the result of how `messageDeleted` handles the removal of the relevant promptId values in the database.

Reply:
> This behaviour is the result of how messageDeleted handles the removal of the relevant promptId values in the database.

Awesome! Any reason the behaviour was changed? I used to have messageDeleted drop the promptId from the database so it could find the details. It would also tie in nicely so an admin could delete spammed prompts and have them closed from the database.

Reply:
Other than the new case handling, the behaviour did not change. [Previously [1]](https://github.com/TechUnderTwenty/TU20Bot/blob/4ba5f0f20b7c2abc9e066b8f97de8a61663d7389/TU20Bot/Handler.cs#L189-L195), when a prompt deletion was detected `messageDeleted` would identify the event-index model by matching the deleted message id with a promptId, and would set the promptId to null if it a match was successful. The change I introduced handles the cases for event-index cancellation and arbitrary prompt message deletion. For instance, when an admin deletes a prompt, the behaviour will now cancel the pending event-index.

[1] https://github.com/TechUnderTwenty/TU20Bot/blob/4ba5f0f20b7c2abc9e066b8f97de8a61663d7389/TU20Bot/Handler.cs#L189-L195

Reply:
Hmm... so it's to differentiate between "the prompt was deleted and you're good to commit everything to index" and "the prompt was deleted and you should wipe any trace of it from the database?"

Reply:
Yes that's right. It represents the state of the current event-index and is used in this implementation to differentiate the readiness of the incoming event to handle the action cleanup.